kiwi/lib/search.py

```diff
@@ -0,0 +1,655 @@
+            and self.config.options.sentence_loss_weight
+        ):
+            if hter:
+                sentence_loss_weight = get_suggestion(
+                    trial,
+                    'sentence_loss_weight',
+                    self.config.options.sentence_loss_weight,
+                )
+                base_config_dict['system']['model']['outputs'][
+                    'sentence_loss_weight'
+                ] = sentence_loss_weight
+                search_values['sentence_loss_weight'] = sentence_loss_weight
+
+        if self.config.options.class_weights and word_level_config:
+            # if self.config.options.search_word_level:
+            #     if not word_level:
+            #         # Word level disabled; nothing to search over
+            #         pass
+            if (
+                self.config.options.class_weights.target_tags
+                and word_level_config['target']
+            ):
+                class_weight_target_tags = get_suggestion(
+                    trial,
+                    'class_weight_target_tags',
+                    self.config.options.class_weights.target_tags,
+                )
+                base_config_dict['system']['model']['outputs']['word_level'][
+                    'class_weights'
+                ]['target_tags'] = {const.BAD: class_weight_target_tags}
+                search_values['class_weight_target_tags'] = class_weight_target_tags
+            if self.config.options.class_weights.gap_tags and word_level_config['gaps']:
+                class_weight_gap_tags = get_suggestion(
+                    trial,
+                    'class_weight_gap_tags',
+                    self.config.options.class_weights.gap_tags,
+                )
+                base_config_dict['system']['model']['outputs']['word_level'][
+                    'class_weights'
+                ]['gap_tags'] = {const.BAD: class_weight_gap_tags}
+                search_values['class_weight_gap_tags'] = class_weight_gap_tags
+            if (
+                self.config.options.class_weights.source_tags
+                and word_level_config['source']
+            ):
+                class_weight_source_tags = get_suggestion(
+                    trial,
+                    'class_weight_source_tags',
+                    self.config.options.class_weights.source_tags,
+                )
+                base_config_dict['system']['model']['outputs']['word_level'][
+                    'class_weights'
+                ]['source_tags'] = {const.BAD: class_weight_source_tags}
+                search_values['class_weight_source_tags'] = class_weight_source_tags
+        return train.Configuration(**base_config_dict), search_values
+
+    def __call__(self, trial) -> float:
+        # TODO: integrate PTL trainer callback
+        # ptl_callback = PyTorchLightningPruningCallback(trial, monitor='val_PEARSON')
+        # train_info = train.run(trainconfig, ptl_callback=ptl_callback)
```

Comment:
I would remove this TODO. I'd rather add a note saying we tried it (I did, there's a branch with it in our private repo) and it works worse than EarlyStopping.

Reply:
Done